tls/s2n_handshake_io.c

```diff
@@ -306,10 +306,7 @@ int s2n_conn_set_handshake_type(struct s2n_connection *conn)
 
 static int s2n_conn_update_handshake_hashes(struct s2n_connection *conn, struct s2n_blob *data)
 {
-    if (s2n_hash_is_available(S2N_HASH_MD5)) {
-        /* The MD5 hash cannot be initialized when FIPS mode is set. */
-        GUARD(s2n_hash_update(&conn->handshake.md5, data->data, data->size));
-    }
+    GUARD(s2n_hash_update(&conn->handshake.md5, data->data, data->size));
```

Comment:
Why was this check removed?

Reply:
`s2n_hash_is_available()` will fail for this MD5 `2n_hash_state` (by design), but it is ignored as this `s2n_hash_state` is used for the PRF. MD5 is permitted in FIPS mode only for the PRF.

Reply:
I think a one line comment explaining that would be useful. I can see myself looking at this 6 months from now confused on how this is "FIPS" compliant.

Reply:
Sounds good. I'll add a comment similar to the comment [where this hash state is initialized](https://github.com/awslabs/s2n/pull/579/files#diff-beda8708c5aaddb196aa5929930b2f01R84).